platform/ontology/api.go

```diff
@@ -55,6 +55,52 @@ func (p *Platform) GetTokenTxsByAddress(address string, token string) (blockatla
 	return txs, nil
 }
 
+func (p *Platform) CurrentBlockNumber() (int64, error) {
+	block, err := p.client.CurrentBlockNumber()
+	if err != nil {
+		logger.Error("CurrentBlockNumber", logger.Params{"platform": p.Coin().Symbol, "details": err.Error()})
+		return 0, err
+	}
+	var height int64
+	if block.Error != 0 {
+		err = errors.E("explorer error")
+	}
+	if len(block.Result) > 0 {
+		height = (int64)(block.Result[0].Height)
+	}
+	return height, nil
+}
+
+func (p *Platform) GetBlockByNumber(num int64) (*blockatlas.Block, error) {
+	response, err := p.client.GetBlockByNumber(num)
+	if err != nil {
+		logger.Error("GetBlockByNumber", logger.Params{"platform": p.Coin().Symbol, "details": err.Error()})
+		return nil, err
+	}
+	var (
+		block blockatlas.Block
+		txs   []blockatlas.Tx
+	)
+	if response.Error == 0 {
+		block.ID = response.Result.Hash
+		block.Number = int64(response.Result.Height)
+		for _, txn := range response.Result.TxnList {
+			tx := new(blockatlas.Tx)
```

Comment:
should part of the code should be moved into `Normalize()` so you can easily test that

Reply:
@EnoRage https://github.com/trustwallet/blockatlas/issues/763
can you handle that, please?